core/src/Kokkos_View.hpp

```diff
@@ -1403,14 +1390,19 @@ class View : public ViewTraits<DataType, Properties...> {
         .template get_label<typename traits::memory_space>();
   }
 
+ private:
+  enum class check_input_args : bool { yes = true, no = false };
+
+ public:
   //----------------------------------------
   // Allocation according to allocation properties and array layout
 
   template <class... P>
   explicit inline View(
       const Impl::ViewCtorProp<P...>& arg_prop,
       std::enable_if_t<!Impl::ViewCtorProp<P...>::has_pointer,
-                       typename traits::array_layout> const& arg_layout)
+                       typename traits::array_layout> const& arg_layout,
+      check_input_args check_args = check_input_args::no)
```

Comment:
The idea here is that we catch discrepancies originating from user error. Other code paths exist, such as 
```
create_mirror_view(const Space&, const Kokkos::View<T, P...>& src,
                   const I&... arg_prop) {
  return typename Impl::MirrorViewType<Space, T, P...>::view_type(
      Kokkos::view_alloc(src.label(), arg_prop...), src.layout());
```
where we instatiate a view of a given type (for example int **, LayoutLeft) and where in that case  the src.layout has a layout of {20, 10, 1, 1, 1, 1, 1}. In that case, the check would fail with "Number of arguments = 8 but dynamic rank = 2". So I think this is ok.